Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYWebCheckoutPaymentProvider.m

```diff
@@ -0,0 +1,193 @@
+//
+//  BUYWebCheckoutPaymentProvider.m
+//  Mobile Buy SDK
+//
+//  Created by Shopify.
+//  Copyright (c) 2015 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "BUYWebCheckoutPaymentProvider.h"
+#import "BUYCheckout.h"
+#import "BUYClient+Checkout.h"
+
+@import SafariServices;
+
+NSString * BUYSafariCallbackURLNotification = @"kBUYSafariCallbackURLNotification";
+NSString * BUYURLKey = @"url";
+NSString * const BUYWebPaymentProviderId = @"BUYWebPaymentProviderId";
+
+static NSString *const WebCheckoutCustomerAccessToken = @"customer_access_token";
+
+@interface BUYWebCheckoutPaymentProvider () <SFSafariViewControllerDelegate>
+@property (nonatomic, strong) BUYCheckout *checkout;
+@property (nonatomic, strong) BUYClient *client;
+@end
+
+@implementation BUYWebCheckoutPaymentProvider
+
+@synthesize delegate;
+
+- (instancetype)initWithClient:(BUYClient *)client
+{
+	NSParameterAssert(client);
+	
+	self = [super init];
+	
+	if (self) {
+		_client = client;
+	}
+	
+	return self;
+}
+
+- (BOOL)isInProgress
+{
+	return (self.checkout != nil);
+}
+
+- (NSUInteger)hash
+{
+	return self.identifier.hash;
+}
+
+- (BOOL)isEqual:(id)object
+{
+	return ([object isKindOfClass:[self class]] && [self.identifier isEqual:[object identifier]]);
+}
+
+- (NSString *)identifier
+{
+	return BUYWebPaymentProviderId;
+}
+
+- (void)startCheckout:(BUYCheckout *)checkout
+{	
+	if (self.isInProgress) {
+		return;
```

Comment:
Should we call the error delegate method, and/or log an error here?  Im thinking the delegate may not be the right use here since it may be interpreted that the original checkout failed.